src/containers/AdminIncomingMessageList/index.tsx

```diff
@@ -63,34 +71,56 @@ const AdminIncomingMessageList: React.FC<AdminIncomingMessageListProps> = (
       }
     : initialContactsFilter;
 
-  const [page, setPage] = useState<number>(0);
-  const [pageSize, setPageSize] = useState<number>(10);
-  const [campaignsFilter, setCampaignsFilter] = useState<any>(
-    initialCampaignsFilter
+  // Query Params
+  const [assignmentsFilter, setAssignmentsFilter] = useQueryParam(
+    "assignmentsFilter",
+    withDefault(AssignmentsFilterParam, initialAssignmentsFilter)
+  );
+  const [campaignsFilter, setCampaignsFilter] = useQueryParam(
+    "campaignsFilter",
+    withDefault(CampaignsFilterParam, initialCampaignsFilter)
+  );
+  const [contactsFilter, setContactsFilter] = useQueryParam(
+    "contactsFilter",
+    withDefault(ContactsFilterParam, defaultContactsFilter)
   );
-  const [contactsFilter, setContactsFilter] = useState<any>(
-    defaultContactsFilter
+  const [tagsFilter, setTagsFilter] = useQueryParam(
+    "tagsFilter",
+    withDefault(TagsFilterParam, defaultTagsFilter)
   );
-  const [assignmentsFilter, setAssignmentsFilter] = useState<any>(
-    initialAssignmentsFilter
+  const [contactNameFilter, setContactNameFilter] = useQueryParam(
+    "nameFilter",
+    ContactNameParam
+  );
+
+  const [page, setPage] = useQueryParam("page", withDefault(NumberParam, 0));
+  const [pageSize, setPageSize] = useQueryParam(
+    "pageSize",
+    withDefault(NumberParam, 10)
   );
-  const [tagsFilter, setTagsFilter] = useState<any>(defaultTagsFilter);
-  const [contactNameFilter, setContactNameFilter] = useState<any>(undefined);
+
+  // Get or Compute Default states
+  const defaultArchivedCampaigns = campaignsFilter.isArchived ?? false;
+  const defaultActiveCampaigns = !campaignsFilter.isArchived ?? true;
+  const defaultIncludeNotOptedOut = contactsFilter.isOptedOut ?? false;
+  const defaultIncludeOptedOut = !contactsFilter.isOptedOut ?? true;
```

Comment:
It evaluates based on when the component is loaded, since this computes the default value, and then the default value is used in `useState` which doesn't change even when filter changes, but only when we change the state which is done by user toggling. 

useQueryParams will read the URL value, or set a default value that we gave to it (without updating the URLs), so if it reads the URL to find a value, the campaignsFilter will be what the URL is, if not, the campaignsFilter will be the default we've given.